src/dcp/dcp.c

```diff
@@ -346,6 +359,291 @@ static int daos_any_error(int rank, bool local_daos_error, int flag_daos_args)
+                d_sg_list_t     sgl;
+                d_iov_t         iov;
+                daos_key_desc_t akey_kds[ENUM_DESC_NR]       = {0};
+                uint32_t        akey_number                  = ENUM_DESC_NR;
+                char            akey_enum_buf[ENUM_DESC_BUF] = {0};
+                char 		akey[ENUM_KEY_BUF] 	     = {0};
+
+                sgl.sg_nr     = 1;
+                sgl.sg_nr_out = 0;
+                sgl.sg_iovs   = &iov;
+
+                d_iov_set(&iov, akey_enum_buf, ENUM_DESC_BUF);
+
+                /* get akeys */
+                rc = daos_obj_list_akey(*src_oh, DAOS_TX_NONE, &diov, &akey_number, akey_kds,
+                                        &sgl, &akey_anchor, NULL);
+                if (rc)
+                    return -1;       
+
+                /* if no akeys returned move on */
+                if (akey_number == 0)
+                    continue;
+
+                int j;
+                char* ptr;
+
+                /* parse out individual akeys based on key length and numver of dkeys returned */
+                for (ptr = akey_enum_buf, j = 0; j < akey_number; j++) {
+                    daos_key_t aiov;
+                    daos_iod_t iod;
+                    snprintf(akey, akey_kds[j].kd_key_len + 1, "%s", ptr);
+                    d_iov_set(&aiov, (void*)akey, akey_kds[j].kd_key_len);
+
+                    /* set iod values */
+                    iod.iod_nr   = 1;
+                    iod.iod_type = DAOS_IOD_SINGLE;
+                    iod.iod_size = DAOS_REC_ANY;
+
+                    d_iov_set(&iod.iod_name, (void*)akey, strlen(akey));
+
+                    /* Do a fetch (with NULL sgl) of single value type, and if that
+		     * returns iod_size == 0, then a single value does not exist. */
+                    rc = daos_obj_fetch(*src_oh, DAOS_TX_NONE, 0, &diov, 1, &iod, NULL, NULL, NULL);
+
+                    /* if iod_size == 0 then this is a DAOS_IOD_ARRAY type */
+                    if ((int)iod.iod_size == 0) {
+                        rc = daos_copy_recx_array(&diov, &aiov, src_oh, dst_oh, &iod);
+                    } else {
+                        rc = daos_copy_recx_single(&diov, src_oh, dst_oh, &iod);
+                    }
+
+                    /* advance to next akey returned */	
+                    ptr += akey_kds[j].kd_key_len;
+                }
+            }
+        }
+    }
+    return rc;
+}
+
+static int daos_obj_copy(daos_args_t* da,
+                          flist_t* flist) {
+    int rc;
+    uint64_t i;
+    const elem_t* p = flist->list_head;
+    for (i = 0; i < flist->list_count; i++) {
+        /* open DAOS object based on oid[i] to get obj handle */
+        daos_handle_t oh;
+        daos_obj_id_t oid;
+        oid.lo = p->obj_id_lo;	
+        oid.hi = p->obj_id_hi;	
+        rc = daos_obj_open(da->src_coh, oid, 0, &oh, NULL);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "DAOS object open returned with errors: ", MFU_ERRF,
```

Comment:
Should we continue if this fails?
Also, I think much of the error handling is probably similar to the DAOS PR you submitted, so I'll stop pointing those out for now, since it's probably redundant. I imagine we'll want a way to centralize these functions anyway. (at least the ones we can)